### PR TITLE
net: openthread: add Link Metrics API

### PR DESCRIPTION
**Pull request by lmaciejonczyk**
This commit implements the OpenThread APIs to configure Enhanced-ACK Based Probing in radio for a specific Initiator. This is needed for Link Metrics functionality.

### CODE REVIEW

File: subsys/net/l2/openthread/Kconfig.features [hunk 1]
```diff
@@ -182,3 +182,6 @@ config OPENTHREAD_OTNS
 
 config OPENTHREAD_FULL_LOGS
 	bool "Enable OpenThread full logs"
+
+config OPENTHREAD_CONFIG_MLE_LINK_METRICS_ENABLE
```
Comment (edmont) [inline subsys/net/l2/openthread/Kconfig.features:186]:
```suggestion
config OPENTHREAD_MLE_LINK_METRICS_ENABLE
```
Same way as with other configurations.

File: include/net/ieee802154_radio.h [hunk 2]
```diff
@@ -208,6 +213,16 @@ struct ieee802154_config {
 			uint32_t start;
 			uint32_t duration;
 		} rx_slot;
+
+		/** ``IEEE802154_CONFIG_ENH_ACK_PROBING`` */
+		struct {
+			bool lqi : 1;
```
Comment (edmont) [inline include/net/ieee802154_radio.h:219]: Thread specification requires only 1 or 2 of among (LQI, LM, RSSI)  to be selected, and order is to be kept. This structure doesn't warranty those requirements. However, OpenThread implementation neither, so maybe such improvement can be added in the future.

File: include/net/ieee802154_radio.h [hunk 3]
```diff
@@ -208,6 +213,16 @@ struct ieee802154_config {
...
+			bool link_margin : 1;
+			bool rssi : 1;
+			uint16_t short_addr;
+			const uint8_t *ext_addr;
+
```
Comment (edmont) [inline include/net/ieee802154_radio.h:224]: nit: unnecessary blank line.

**Review by edmont — APPROVED** [review]

File: include/net/ieee802154_radio.h [hunk 4]
```diff
@@ -162,6 +162,11 @@ enum ieee802154_config_type {
 
 	/** Configure a radio reception slot */
 	IEEE802154_CONFIG_RX_SLOT,
+
+	/** Enable/disable or update Enhanced-ACK Based Probing in radio
+	 *  for a specific Initiator.
+	 */
+	IEEE802154_CONFIG_ENH_ACK_PROBING,
```
Comment (rlubos) [inline include/net/ieee802154_radio.h:169]: It's a bit unclear to me, OpenThread enables probing, and what then? Will this trigger some callbacks from the radio side?

Comment (edmont) [reply include/net/ieee802154_radio.h:169]: From this moment the device should include the configured link metrics values in every Enh-ACK generated for this neighbor.

Comment (rlubos) [reply include/net/ieee802154_radio.h:169]: Ah, so that's how it works, thanks for clarifying.

**Review by rlubos — APPROVED** [review]

**Comment by galak** [comment]
@lmaciejonczyk can you rebase to address the merge conflicts.

**Comment by aurel32** [comment]
On the `openthread` module, there is zephyr/CMakeLists.txt:

```
if(CONFIG_OPENTHREAD_LINK_METRICS)
  set(OT_LINK_METRICS ON CACHE BOOL "Enable Link Metrics feature for Thread 1.2")
endif()
```

`OT_LINK_METRICS` basically sets `OPENTHREAD_CONFIG_MLE_LINK_METRICS_ENABLE=1`, so it seems to be a different way to enable Link Metrics. However there is nothing that allows to enable `CONFIG_OPENTHREAD_LINK_METRICS` on the Zephyr side.

**Comment by lmaciejonczyk** [comment]
> On the `openthread` module, there is zephyr/CMakeLists.txt:
> 
> ```
> if(CONFIG_OPENTHREAD_LINK_METRICS)
>   set(OT_LINK_METRICS ON CACHE BOOL "Enable Link Metrics feature for Thread 1.2")
> endif()
> ```
> 
> `OT_LINK_METRICS` basically sets `OPENTHREAD_CONFIG_MLE_LINK_METRICS_ENABLE=1`, so it seems to be a different way to enable Link Metrics. However there is nothing that allows to enable `CONFIG_OPENTHREAD_LINK_METRICS` on the Zephyr side.

Good catch, thanks!
Will be fixed by #35035

**Pull request merged**